### ISSUE TITLE
Feature: Scenes

### ISSUE THREAD

**Issue by ceremcem** [realthunder/FreeCAD_assembly3#186]
# Problem

While a complex assembly is being created, it becomes necessary to focus to a part *quite often* by hiding the rest of the model. When we are done with that part, we need to show everything that we had hidden previously. This becomes a burden after a point.

# Proposal

Like the [SketchUp's "add scene" feature](https://www.youtube.com/watch?v=EOeraP8K544), it might be very useful to be able to add scenes that contain one (usually one) or more parts which are to be shown and the rest are hidden while being able to switch between those scenes on demand and switch back to "whole scene". 

![sketchup-scenes](https://user-images.githubusercontent.com/6639874/51873400-13f97100-236e-11e9-89fe-acefdb2832ea.gif)

It would be great if scenes were support (ordered by importance): 
* Which parts to be shown in that specific scene
* Camera angles
* Transparency of hidden parts (do we want to see only the selected parts while rest are completely hidden or we also want to see the "hidden" parts *very* transparently)

**Comment by realthunder** [comment]
Yeah, this is also something I want. I believe the STEP standard also support this type of thing.
